File: system-test/dotnet/Driver.Adapter/External/Erp/Client/ErpStubClient.cs
using Common;

using Driver.Adapter.Shared.Client.WireMock;

using Driver.Adapter.Shared.Client.Http;

using Driver.Adapter.External.Erp.Client.Dtos;

using Driver.Adapter.External.Erp.Client.Dtos.Error;



namespace Driver.Adapter.External.Erp.Client;



public class ErpStubClient : BaseErpClient

{

    private const string ErpProductsEndpoint = "/erp/api/products";
    private const string ErpPromotionEndpoint = "/erp/api/promotion";



    private readonly JsonWireMockClient _wireMockClient;



    public ErpStubClient(string baseUrl) : base(baseUrl)

    {

        _wireMockClient = new JsonWireMockClient(baseUrl);

    }



    public new void Dispose()

    {

        base.Dispose();

        _wireMockClient?.Dispose();

    }



    public Task<Result<VoidValue, ExtErpErrorResponse>> ConfigureGetProductAsync(ExtProductDetailsResponse response)

        => _wireMockClient.StubGetAsync($"{ErpProductsEndpoint}/{response.Id}", HttpStatus.Ok, response)

            .MapErrorAsync(ExtErpErrorResponse.From);

    public Task<Result<VoidValue, ExtErpErrorResponse>> ConfigureGetPromotionAsync(ExtGetPromotionResponse response)

        => _wireMockClient.StubGetAsync(ErpPromotionEndpoint, HttpStatus.Ok, response)

            .MapErrorAsync(ExtErpErrorResponse.From);

    public Task RemoveStubsAsync() => _wireMockClient.RemoveStubsAsync();

}

File: system-test/java/src/main/java/com/optivem/shop/dsl/driver/adapter/external/erp/client/ErpStubClient.java
package com.optivem.shop.dsl.driver.adapter.external.erp.client;

import com.optivem.shop.dsl.driver.adapter.shared.client.http.HttpStatus;
import com.optivem.shop.dsl.driver.adapter.external.erp.client.dtos.ExtGetPromotionResponse;
import com.optivem.shop.dsl.driver.adapter.external.erp.client.dtos.ExtProductDetailsResponse;
import com.optivem.shop.dsl.driver.adapter.external.erp.client.dtos.error.ExtErpErrorResponse;
import com.optivem.shop.dsl.common.Result;
import com.optivem.shop.dsl.driver.adapter.shared.client.wiremock.JsonWireMockClient;

public class ErpStubClient extends BaseErpClient {
    private static final String ERP_PRODUCTS_ENDPOINT = "/erp/api/products";
    private static final String ERP_PROMOTION_ENDPOINT = "/erp/api/promotion";

    private final JsonWireMockClient wireMockClient;

    public ErpStubClient(String baseUrl) {
        super(baseUrl);
        this.wireMockClient = new JsonWireMockClient(baseUrl);
    }

    public Result<Void, ExtErpErrorResponse> configureGetProduct(ExtProductDetailsResponse response) {
        var sku = response.getId();
        return wireMockClient.stubGet(ERP_PRODUCTS_ENDPOINT + "/" + sku, HttpStatus.OK, response)
                .mapError(ExtErpErrorResponse::new);
    }

    public Result<Void, ExtErpErrorResponse> configureGetPromotion(ExtGetPromotionResponse response) {
        return wireMockClient.stubGet(ERP_PROMOTION_ENDPOINT, HttpStatus.OK, response)
                .mapError(ExtErpErrorResponse::new);
    }

    public void removeStubs() {
        wireMockClient.removeStubs();
    }
}
